controllers/friends.js

```diff
@@ -5,23 +5,37 @@ const FriendsController = {
     console.log("put request made");
     const friendUsername = req.params.friendUsername;
     const currentUser = req.session.user;
-    const newFriendList = currentUser.friends.concat(friendUsername);
+    let newFriendList;
 
-    // update currentUser's friend list
-    User.updateOne(
-      { username: currentUser.username },
-      { friends: newFriendList },
-      (err) => {
+    User.findOne({ username: friendUsername }, (err, friendUser) => {
+      if (err) {
         // do something if there's an error?
-        if (err) {
-          console.log(err);
-        } else {
-          req.session.user.friends = newFriendList;
-          console.log("user added to friends list");
-          res.redirect(303, "/profiles/" + friendUsername);
-        }
+        console.log(err);
+      } else {
+        // update current user's friend list with new friend added
+        newFriendList = currentUser.friends.concat(friendUser._id);
+        User.updateOne(
+          { username: currentUser.username },
+          { $push: { friends: friendUser._id } },
```

Comment:
We might want to change to $addToSet (this only adds a friend to the list if it doesn't yet exist).

Reply:
good point, I'll try this and see if it works as well